src/state/user/hooks/index.tsx

```diff
@@ -82,6 +84,22 @@ export function useExchangeChartManager(isMobile: boolean): [boolean, (isDisplay
   return [isMobile ? false : isChartDisplayed, setUserChartPreference]
 }
 
+export function useExchangeChartViewManager() {
+  const dispatch = useDispatch<AppDispatch>()
+  const chartViewMode = useSelector<AppState, AppState['user']['userChartViewMode']>(
+    (state) => state.user.userChartViewMode,
+  )
+
+  const setUserChartViewPreference = useCallback(
+    (view: ChartViewMode) => {
+      dispatch(setChartViewMode(view))
+    },
+    [dispatch],
+  )
+
+  return [chartViewMode, setUserChartViewPreference] as const
```

Comment:
Why `as const`? 👀

Reply:
IMO, it is better (or lazy) way to let type inter its array order
<img width="988" alt="CleanShot 2021-12-05 at 13 13 48@2x" src="https://user-images.githubusercontent.com/94336009/144734597-2b273bda-895d-476a-92ca-ff7eadc00ae5.png">